src/controllers/legacycontrollersettings.h

```diff
@@ -6,6 +6,14 @@
 #include "controllers/legacycontrollersettingslayout.h"
 #include "util/parented_ptr.h"
 
+namespace {
+template<class T>
+bool valid_range(T min, T max, T step) {
+    return (min <= 0 && min + step <= max) ||
+            (max >= 0 && max - step >= min);
```

Comment:
```suggestion
    VERIFY_OR_DEBUG_ASSERT(step >= 0) {
        return false;
    }
    return (min <= 0 && min + step <= max) ||
            (max >= 0 && max - step >= min);
```

Reply:
Alternative would be moving the test "step > 0" into this range check as well.